### PR TITLE
feat: cast custom data

### PR DESCRIPTION
**Pull request by cjpillsbury**
* Exposes `castReceiver` / `cast-receiver` prop / attr to specify a custom receiver app
* Adds mux-specific custom data for a custom Google Chromecast receiver app (minimally necessary for DRM license requests).
* Adds `drmType` to Mux Data metadata once the DRM type is determined via key session initialization.
* Relates to https://github.com/muxinc/media-elements/pull/32

Also:
* Ensures omitted storyboard tokens and thumbnail tokens don't cause errors.

### CODE REVIEW

File: packages/mux-player/REFERENCE.md [hunk 1]
```diff
@@ -150,6 +151,8 @@
 | `chapters` <sub><sup>Read only</sup></sub> | `Array<{ startTime: number; endTime?: number, value: string; }>` | The array of Chapters for the current media, added via `addChapters(chapters)`. | `[]` |
 | `activeCuePoint` <sub><sup>Read only</sup></sub> | `{ startTime: number; endTime?: number, value: any; }` | The current active CuePoint, determined based on the player's `currentTime`. | `undefined` |
 | `activeChapter` <sub><sup>Read only</sup></sub> | `{ startTime: number; endTime?: number, value: string; }` | The current active Chapter, determined based on the player's `currentTime`. | `undefined` |
+| `castReceiver` | `string` (Receiver ID) | The app ID to use for a custom [Google cast receiver](https://developers.google.com/cast/docs/web_receiver/basic). If none is provided, the default receiver app will be used. | `undefined` |
+| `castCustomData` | `object` (JSON-serializable) | [Custom Data](https://developers.google.com/cast/docs/reference/web_sender/chrome.cast.media.MediaInfo#customData) to send to your Google cast receiver on initial load. If none is provided, various Mux key/value pairs will be sent. | Mux-specific object |
```
Comment (cjpillsbury) [inline packages/mux-player/REFERENCE.md:155]: Wasn't sure how to document the default value here, which is a bunch of mappings of mux player props to an object

File: packages/mux-player/src/template.ts [hunk 2]
```diff
@@ -128,6 +128,7 @@ export const content = (props: MuxTemplateProps) => html`
       custom-domain="${props.customDomain ?? false}"
       src="${!!props.src ? props.src : props.playbackId ? toMuxVideoURL(props) : false}"
       cast-src="${!!props.src ? props.src : props.playbackId ? toMuxVideoURL(props) : false}"
+      cast-receiver="${!!props.castReceiver ?? false}"
```
Comment (luwes) [inline packages/mux-player/src/template.ts:131]: (blocking) remove double negation?

File: packages/mux-player/src/template.ts [hunk 3]
```diff
@@ -139,7 +140,7 @@ export const content = (props: MuxTemplateProps) => html`
       <media-poster-image
         part="poster"
         exportparts="poster, img"
-        src="${props.poster === '' ? false : props.poster ?? false}"
+        src="${!!props.poster ? props.poster : false}"
```
Comment (luwes) [inline packages/mux-player/src/template.ts:143]: I think it's correct but these are sometimes tricky changes

Comment (cjpillsbury) [reply packages/mux-player/src/template.ts:143]: Yeah pretty sure this one is safe.

File: packages/mux-player/src/template.ts (hunk 2 above)
Comment (cjpillsbury) [reply packages/mux-player/src/template.ts]: Thanks for the catch! (dangers of flying blind with code changes 😅)

**Review by luwes — APPROVED** [review]
👍

**Pull request merged**